frontend/pages/sites/$siteId/builds/index.jsx

```diff
@@ -45,48 +28,34 @@ function SiteBuildList() {
   const organization = useSelector((state) =>
     getOrgById(state.organizations, site.organizationId),
   );
-  const builds = useSelector((state) => state.builds);
 
-  useEffect(() => {
-    buildActions.fetchBuilds({
-      id,
-    });
-
-    const intervalHandle = setInterval(() => {
-      if (document.visibilityState === 'visible') {
-        buildActions.refetchBuilds({ id });
-      }
-    }, REFRESH_INTERVAL);
-    return () => {
-      clearInterval(intervalHandle);
-    };
-  }, []);
-
-  if (!builds?.isLoading && builds?.data?.length === 0) {
-    const header = 'This site does not yet have any builds.';
-    const message = `
-      If this site was just added,
-      the first build should be available within a few minutes.
-    `;
-    return <AlertBanner status="info" header={header} message={message} />;
-  }
-  const latestBuilds = builds.data && latestBuildByBranch(builds.data);
+  const { data, error, isPending } = useBuilds(id);
 
   return (
-    <div className="grid-row">
-      {organization?.isSandbox && (
-        <div className="well">
-          <AlertBanner
-            status="warning"
-            message={sandboxMsg(organization.daysUntilSandboxCleaning, 'site builds')}
-            alertRole={false}
-          />
-        </div>
-      )}
-      {builds?.data?.length > 0 && (
-        <div className="grid-col-12 table-container">
-          <table
-            className={`
+    <QueryPage
+      data={data}
+      dataHeader={'This site does not yet have any builds.'}
+      dataMessage={
+        'If this site was just added, ' +
+        'the first build should be available within a few minutes.'
+      }
+      error={error}
+      isPending={isPending}
+    >
+      <div className="grid-row">
+        {organization?.isSandbox && (
+          <div className="well">
+            <AlertBanner
+              status="warning"
+              message={sandboxMsg(organization.daysUntilSandboxCleaning, 'site builds')}
+              alertRole={false}
+            />
+          </div>
+        )}
+        {data?.length > 0 && (
```

Comment:
Is there a way to avoid these checks in the `QueryPage` children? It shouldn't try to render the child without data but not sure if component still gets fully run

Reply:
This could be cut because the `QueryPage` does not render the child without data.